Doctrine/Orm/DataProvider.php

```diff
@@ -30,13 +30,13 @@ class DataProvider implements DataProviderInterface
     private $managerRegistry;
 
     /**
-     * @var QueryExtensionInterface[]
+     * @var QueryCollectionExtensionInterface[]|QueryItemExtensionInterface[]
```

Comment:
We must have two different methods: `addItemExtension(ItemExtensionInterface $extension)` and `addCollectionExtension(CollectionExtensionInterface $extension)`